### PR TITLE
[Admin] Ill do it myself.. Fixes Delay and makes it not shit

### PR DESCRIPTION
**Pull request by JamieD1**
Makes delay one verb and not shit

#### Changelog

:cl:    
tweak: Admin delay verb is better
/:cl:

### CODE REVIEW

File: code/modules/admin/admin.dm [hunk 1]
```diff
@@ -661,23 +639,47 @@
...
+			else
+				to_chat(world, "<b>The game will start in [DisplayTimeText(newtime)].</b>")
+				SEND_SOUND(world, sound('sound/ai/attention.ogg'))
+				message_admins("[key_name(usr)] set the pre-game delay to [DisplayTimeText(newtime)].")
+				log_admin("[key_name(usr)] set the pre-game delay to [DisplayTimeText(newtime)].")
+			SSblackbox.record_feedback("tally", "admin_verb", 1, "Delay Game Start") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+
+	else if(SSticker.current_state == GAME_STATE_PLAYING)
+		to_chat(usr, "Round is still running, no need to delay.")
+	else if(SSticker.current_state > GAME_STATE_PLAYING)
```
Comment (alexkar598) [inline code/modules/admin/admin.dm]: Theres valid reasons to delay the restart before the round is over

**Review by alexkar598 — CHANGES_REQUESTED** [review]

File: code/modules/admin/admin.dm (hunk 1 above)
Comment (JamieD1) [reply code/modules/admin/admin.dm]:
```suggestion
	else if(SSticker.current_state >= GAME_STATE_PLAYING)
```

File: code/modules/admin/admin.dm [hunk 2]
```diff
@@ -661,23 +639,45 @@
...
+				log_admin("[key_name(usr)] delayed the round start.")
+				message_admins("[key_name(usr)] delayed the round start.")
+			else
+				to_chat(world, "<b>The game will start in [DisplayTimeText(newtime)].</b>")
+				SEND_SOUND(world, sound('sound/ai/attention.ogg'))
+				message_admins("[key_name(usr)] set the pre-game delay to [DisplayTimeText(newtime)].")
+				log_admin("[key_name(usr)] set the pre-game delay to [DisplayTimeText(newtime)].")
+			SSblackbox.record_feedback("tally", "admin_verb", 1, "Delay Game Start") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+
+	else if(SSticker.current_state >= GAME_STATE_PLAYING)
```
Comment (alexkar598) [inline code/modules/admin/admin.dm]: If it aint smaller its bigger or equal innit

```suggestion
	else
```

File: code/modules/admin/admin.dm [hunk 3]
```diff
@@ -661,23 +639,45 @@
...
+		SSticker.delay_end = !SSticker.delay_end
+		var/reason = SSticker.delay_end ? "for reason: [SSticker.admin_delay_notice]" : "."//laziness
+		var/msg = "[SSticker.delay_end ? "delayed" : "undelayed"] the round end [reason]"
+		log_admin("[key_name(usr)] [msg]")
+		message_admins("[key_name_admin(usr)] [msg]")
+		if(SSticker.ready_for_reboot && !SSticker.delay_end) //we undelayed after standard reboot would occur
+			if(alert(usr,"Restart the round?.","Round restart","Yes","No") == "Yes")
+				SSticker.Reboot(delay = 100, force = TRUE)
+	else
+		return alert("Game is in an unexpected state!")
```
Comment (alexkar598) [inline code/modules/admin/admin.dm]: Not needed due to above comment

File: code/modules/admin/admin.dm [hunk 4]
```diff
@@ -661,23 +639,45 @@
...
+			if(isnull(SSticker.admin_delay_notice))
+				return
 		else
-			to_chat(world, "<b>The game will start in [DisplayTimeText(newtime)].</b>")
-			SEND_SOUND(world, sound('sound/ai/attention.ogg'))
-			log_admin("[key_name(usr)] set the pre-game delay to [DisplayTimeText(newtime)].")
-		SSblackbox.record_feedback("tally", "admin_verb", 1, "Delay Game Start") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+			SSticker.admin_delay_notice = null
+		SSticker.delay_end = !SSticker.delay_end
+		var/reason = SSticker.delay_end ? "for reason: [SSticker.admin_delay_notice]" : "."//laziness
```
Comment (alexkar598) [inline code/modules/admin/admin.dm]: ``Having it insert a dot would display X has delayed the round end .`` with that space at the end
```suggestion
		var/reason = SSticker.delay_end ? "for reason: [SSticker.admin_delay_notice]" : ""
```

**Review by alexkar598 — DISMISSED** [review]

**Review by fluffe9911 — APPROVED** [review]
I trust our hc

**Pull request merged**